tests/specs/mcd/verification.k

```diff
@@ -178,12 +212,24 @@ module LEMMAS-MCD-COMMON
     rule (W0 : WS) [ 0     .. WIDTH ] => W0 : (WS [ 0 .. WIDTH -Int 1 ])  requires 0 <Int WIDTH [simplification]
     rule (_  : WS) [ START .. WIDTH ] => WS [ START -Int 1 .. WIDTH ]     requires 0 <Int START [simplification]
 
-    // ### cat-exhaustiveness
-
     rule WS [ START .. WIDTH ] [ 0 .. WIDTH' ]  => WS [ START .. WIDTH' ]  requires WIDTH' <=Int WIDTH [simplification]
 
+    // ### cat-exhaustiveness
+
     rule ( BA1:ByteArray ++ BA2:ByteArray ) ++ BA3:ByteArray => BA1 ++ ( BA2 ++ BA3 ) [simplification]
 
+  // ########################
+  // Keccak
+  // ########################
+
+    rule #lookup( M:Map [ keccak(_) <- _ ], I:Int ) => #lookup( M , I )
+      requires 0 <=Int I andBool I <Int 10
+      [simplification]
+
+    rule #lookup( M:Map [ keccak(_) +Int I1:Int <- _ ], I2:Int ) => #lookup( M , I2 )
+      requires 0 <=Int I1 andBool I1 <Int 10 andBool 0 <=Int I2 andBool I2 <Int 10
+      [simplification]
```

Comment:
Same here. I think we can isolate the "untrueness" of keccak's properties to a bunch of lemmas about disequalities, instead of lifting them through map lookups.

Reply:
Yes, as above.

Reply:
Unresolving until the code shows up here.

Reply:
Should be there now.